utils.h

```diff
@@ -30,6 +30,31 @@ CT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
 #include <time.h>
 #include "types.h"
 
+class ostreamWrapper {
+private:
+    std::ostream* str;
+
+public:
+    ostreamWrapper(std::ostream* str_v) : str(str_v) {}
+
+    template <typename T>
+    ostreamWrapper& operator<<(const T& t) {
+        #if defined(PCM_PRINTS)
```

Comment:
if one disables all cerr output by default, users will miss all error hints (e.g. no permission or driver is missing). Therefore they will not be able to fix it which will result in a lot of confusion. You have to change the logic: disable the error output optionally.